src/Voron/Data/BTrees/Tree.cs

```diff
@@ -30,8 +31,13 @@ public unsafe partial class Tree
 
         private readonly TreeMutableState _state;
 
-        private static readonly ObjectPool<RecentlyFoundTreePages> _currentRecentlyFoundPagesPool = new(() => new RecentlyFoundTreePages(), 128);
-        private RecentlyFoundTreePages _recentlyFoundPages;
+        private static readonly PerCoreStatic<ObjectPool<RecentlyFoundTreePages>> FoundPagesPool = new(
```

Comment:
I'm a bit worried about that change because of _potential_ problems related to similar change we made in https://github.com/ravendb/ravendb/pull/17946.

Let me hold on with merging this PR until we'll know if that change was indeed the source of https://issues.hibernatingrhinos.com/issue/RavenDB-21521

Reply:
Agree, this can only point us into improper usage of the underlying object state (most likely) or a failure of the thread safety of `ArrayPool` (unlikely and very bad).

Reply:
Given that `PerCoreStatic` is highly suspicious can we revert this part only so we'll be able to merge this PR?